homeassistant/components/mqtt/__init__.py

```diff
@@ -264,7 +266,19 @@ def validate_device_has_at_least_one_identifier(value: ConfigType) -> \
 # pylint: disable=invalid-name
 PublishPayloadType = Union[str, bytes, int, float, None]
 SubscribePayloadType = Union[str, bytes]  # Only bytes if encoding is None
-MessageCallbackType = Callable[[str, SubscribePayloadType, int], None]
+
+
+@attr.s(slots=True, frozen=True)
+class Message:
+    """MQTT Message."""
+
+    topic = attr.ib(type=str)
+    payload = attr.ib(type=PublishPayloadType)
+    qos = attr.ib(type=int, default=0)
```

Comment:
Changed.
`Message` class was already used to hold will and birth messages, but it should be fine to change since will and birth conf schema has default value for qos and retain.